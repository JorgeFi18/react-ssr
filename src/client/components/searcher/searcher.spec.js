import React from 'react';
import renderer from 'react-test-renderer';
import Searcher from './searcher.component';

describe('Searcher component', () => {
    it('Should enter hello text', () => {
        let text = '';
        const changeText = e => {text = e};

        const component = renderer.create(
            <Searcher onSearchChange={changeText}/>
        )

        let tree = component.toJSON();
        const [ input ] = tree.children;
        input.props.onChange('hello');

        expect(text).toBe('hello');
    });

    it('Should check input placeholder', ()=> {
        const component = renderer.create(
            <Searcher />
        )

        let tree = component.toJSON();
        const [ input ] = tree.children;

        expect(input.props.placeholder).not.toBe('coconut');
    });

    it('Should match with snapshot', ()=> {
        const component = renderer.create(
            <Searcher />
        )

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});