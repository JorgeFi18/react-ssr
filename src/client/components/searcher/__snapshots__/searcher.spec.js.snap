// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searcher component Should match with snapshot 1`] = `
<div
  className="searcher"
>
  <input
    className="searcher__input"
    data-cy="search"
    name="search"
    placeholder="Search for your character"
    type="search"
  />
</div>
`;
